### PR TITLE
bug fix for 158 and 160

### PR DESCRIPTION
**Pull request by ar-india**
Signed-off-by: ar-india <alokranjan2005@gmail.com>
https://nimblejira.nimblestorage.com/browse/SA-160
https://nimblejira.nimblestorage.com/browse/SA-158

### CODE REVIEW

File: ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_fc.py [hunk 1]
```diff
@@ -133,18 +133,18 @@ def update_fc_interface(
         online):
 
     if utils.is_null_or_empty(array_name_or_serial):
-        return (False, False, "Fibre channel interface update failed as no array name is provided.", {})
+        return (False, False, "Fibre channel interface update failed as no array name is provided.", {}, {})
```
Comment (ar-india) [inline ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_fc.py:136]: FC was not returning attrs.. made changes for that

File: ansible_collection/hpe/nimble/plugins/module_utils/hpe_nimble.py [hunk 2]
```diff
@@ -306,3 +306,24 @@ def get_initiator_group_id(client_obj, ig_name):
...
+        return False
+
+    temp = array_version.split('-')
+    array_version = temp[0]
+    arr_version = array_version.split('.')
+    version_to_check = arr_version_to_check.split('.')
+    if arr_version[0] > version_to_check[0]:
+        return True
+    elif arr_version[0] >= version_to_check[0] and arr_version[1] >= version_to_check[1]:
+        return True
```
Comment (rgcostea) [inline ansible_collection/hpe/nimble/plugins/module_utils/hpe_nimble.py:327]: Is it enough to only check major.minor?  We should probably check major.minor.maintenance.

File: ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_info.py [hunk 3]
```diff
@@ -802,8 +799,13 @@ def fetch_minimum_subset(info_subset):
             if key == 'arrays':
                 resp = cl_obj.list(detail=True, fields="extended_model,full_name,all_flash")
             elif key == 'groups':
-                resp = cl_obj.list(detail=True,
-                                   fields="encryption_config,name,fc_enabled,iscsi_enabled,leader_array_name,default_iscsi_target_scope,num_snaps")
+                cl_obj = info_subset['arrays']
+                # certainn fields were only added in NimOS 5.1 and above
```
Comment (rgcostea) [inline ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_info.py]: certain

**Review by rgcostea — APPROVED** [review]
lgtm

File: ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_info.py (hunk 3 above)
Comment (ar-india) [reply ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_info.py]: done

File: ansible_collection/hpe/nimble/plugins/module_utils/hpe_nimble.py (hunk 2 above)
Comment (ar-india) [reply ansible_collection/hpe/nimble/plugins/module_utils/hpe_nimble.py:327]: have modified the code to check for major.minor.mainten..

**Pull request merged**